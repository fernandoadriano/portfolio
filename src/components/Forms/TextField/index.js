import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Text from '../../../foundations/typography/Text';
import { getThemeColor } from '../../../theme';

const InputWrapper = styled.div`
  margin-bottom: 17px;
`;

const Input = styled(Text)`
  width: 100%;
  border: 2px solid ${({ theme }) => theme.colors.primary.main};
  padding: 8px 14px;
  outline: 0;
  color: ${({ color, theme }) => getThemeColor(color, theme)};
  border-radius: ${({ rounded, theme }) => (rounded ? theme.borderRadius : '0px')};
  &::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
    color: ${({ placeholderColor, theme }) => getThemeColor(placeholderColor, theme)};
    opacity: 1; /* Firefox */
  }  
`;

Input.defaultProps = {
  as: 'input',
  variant: 'paragraph1',
};

export default function TextField({
  placeholder,
  name,
  onChange,
  value,
  rounded,
  color,
  placeholderColor,
  as,
  rows,
}) {
  return (
    <InputWrapper>
      <Input
        as={as}
        type="text"
        placeholder={placeholder}
        name={name}
        onChange={onChange}
        value={value}
        rounded={rounded}
        color={color}
        placeholderColor={placeholderColor}
        rows={rows}
      />
    </InputWrapper>
  );
}

TextField.defaultProps = {
  rounded: false,
  color: null,
  placeholderColor: null,
  as: 'input',
  rows: 1,
};

TextField.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  rounded: PropTypes.bool,
  color: PropTypes.string,
  placeholderColor: PropTypes.string,
  as: PropTypes.string,
  rows: PropTypes.number,
};
