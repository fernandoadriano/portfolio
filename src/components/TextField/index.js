/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Text from 'src/foundations/typography/Text';
import { getThemeColor } from 'src/theme';

// const InputWrapper = styled.div`
//   margin-bottom: 17px;
// `;

const InputWrapper = styled.input`
  width: 100%;
  border: 2px solid ${({ theme }) => theme.colors.primary.main};
  padding: 8px 14px;
  outline: 0;
  color: ${({ color, theme }) => getThemeColor(color, theme)};
  border-radius: ${({ rounded, theme }) => (rounded ? theme.borderRadius : '0px')};
  &::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
    color: ${({ placeholderColor, theme }) => getThemeColor(placeholderColor, theme)};
    opacity: 1; /* Firefox */
  }  
`;

const Input = ({
  as, variant, autoFocus, ...props
}) => (
  <InputWrapper
    as={as}
    variant={variant}
    autoFocus={autoFocus}
    {...props}
  />
);

Input.defaultProps = {
  as: 'input',
  variant: 'paragraph1',
  autoFocus: false,
};

Input.propTypes = {
  as: PropTypes.string,
  autoFocus: PropTypes.bool,
  variant: PropTypes.string,
};
export default function TextField({
  placeholder,
  name,
  onChange,
  value,
  rounded,
  color,
  placeholderColor,
  as,
  autoFocus,
  rows,
}) {
  return (
    <Input
      as={as}
      autoFocus={autoFocus}
      type="text"
      placeholder={placeholder}
      name={name}
      onChange={onChange}
      value={value}
      rounded={rounded}
      color={color}
      placeholderColor={placeholderColor}
      rows={rows}
    />
  );
}

TextField.defaultProps = {
  autoFocus: false,
  rounded: false,
  color: null,
  placeholderColor: null,
  as: 'input',
  rows: 1,
};

TextField.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  rounded: PropTypes.bool,
  autoFocus: PropTypes.bool,
  color: PropTypes.string,
  placeholderColor: PropTypes.string,
  as: PropTypes.string,
  rows: PropTypes.number,
};
