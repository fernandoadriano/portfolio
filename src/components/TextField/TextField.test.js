import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from 'src/infra/test/testUtils';

import TextField from './index';

describe('<Input />', () => {
  describe('Renderização', () => {
    it('Campo padrão', () => {
      const valorInicial = 'valor inicial';
      render(
        <TextField
          name="nome"
          placeholder="nome"
          onChange={() => {}}
          value={valorInicial}
          color="red"
          placeholderColor="blue"
        />,
      );

      const input = screen.getByPlaceholderText('nome');

      expect(input).toHaveProperty('value', valorInicial);
      expect(input).toHaveProperty('name', 'nome');
      expect(input).toMatchSnapshot();
    });
  });
});
