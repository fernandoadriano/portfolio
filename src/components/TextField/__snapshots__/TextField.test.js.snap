// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renderização Campo padrão 1`] = `
.c0 {
  width: 100%;
  border: 2px solid #138AF3;
  padding: 8px 14px;
  outline: 0;
  border-radius: 0px;
}

.c0::-webkit-input-placeholder {
  opacity: 1;
}

.c0::-moz-placeholder {
  opacity: 1;
}

.c0:-ms-input-placeholder {
  opacity: 1;
}

.c0::placeholder {
  opacity: 1;
}

<input
  class="c0"
  color="red"
  name="nome"
  placeholder="nome"
  rows="1"
  type="text"
  value="valor inicial"
/>
`;
